src/freshrssurlreader.cpp

```diff
@@ -0,0 +1,79 @@
+#include "freshrssurlreader.h"
+
+#include "configcontainer.h"
+#include "fileurlreader.h"
+#include "logger.h"
+#include "remoteapi.h"
+#include "utils.h"
+
+namespace newsboat {
+
+FreshRssUrlReader::FreshRssUrlReader(ConfigContainer* c,
+	const std::string& url_file,
+	RemoteApi* a)
+	: cfg(c)
+	, file(url_file)
+	, api(a)
+{
+}
+
+FreshRssUrlReader::~FreshRssUrlReader() {}
+
+nonstd::optional<std::string> FreshRssUrlReader::reload()
+{
+	urls.clear();
+	tags.clear();
+	alltags.clear();
+
+	if (cfg->get_configvalue_as_bool("freshrss-show-special-feeds")) {
+		std::vector<std::string> tmptags;
+		tmptags.clear();
+		urls.push_back((cfg->get_configvalue("freshrss-url") +
+				"/reader/api/0/stream/contents/user/-/state/com.google/starred"));
+		tmptags.push_back(std::string("~") + _("Starred items"));
+		tags[(cfg->get_configvalue("freshrss-url") +
+								      "/reader/api/0/stream/contents/user/-/state/com.google/starred")] = tmptags;
```

Comment:
Since I already requested a couple changes… I bet you can clean this up a bit:

1. no need to `clear()` a `vector` that was just created;
2. looks like the whole `get_configvalue() + "..."` thing could be extracted into a local `const std::string` and reused? It'd certainly read better.